views/project/src/components/EDATargetCheckbox.vue

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+import type {
+  YosysOptions,
+  NextpnrOptions
+} from 'edacation';
+import { defineComponent } from 'vue';
+
+import { state as globalState } from '../state';
+
+export default defineComponent({
+  props: {
+    worker: {
+      type: String,
+      required: true
+    },
+    configTarget: {
+      type: String,
+      required: true
+    },
+    label: {
+      type: String
+    }
+  },
+  data() {
+    return {
+      state: globalState,
+      option: false
+    };
+  },
+  methods: {
+    toggleOption() {
+      this.option = !this.option;
+
+      const { project } = this.state;
+      if (!project) return;
+
+      project.configuration = project.configuration || {};
+      project.configuration.defaults = project.configuration.defaults || {};
```

Comment:
Your code currently only works for the `defaults` section, but it should also work for other targets. You can look at how `EDATargetValueList` uses `targetIndex`.

Reply:
How can I see if it changed other targets? I pushed another version which uses the targetIndex but I haven't been able to test it because I am not sure how